### PR TITLE
added creditorReferenceInformation

### PR DESCRIPTION
**Pull request by SevenbitsIt**
Added the possibility to add a reference number given by the creditor. This number is usually found on Invoices so that the beneficial company can assign the payment to a special business case or invoice.

### CODE REVIEW

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php [hunk 1]
```diff
@@ -61,8 +61,15 @@ public function addTransfer($paymentName, array $transferInformation)
             $transferInformation['creditorName']
         );
         $transfer->setBic($transferInformation['creditorBic']);
-        $transfer->setRemittanceInformation($transferInformation['remittanceInformation']);
-
+        
+        if(array_key_exists('creditorReferenceInformation', $transferInformation) &&
+            strlen($transferInformation['creditorReferenceInformation'])>5 ) {
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: What is this constraint (>5) for  ? and if really usefull can you create a constant from ?

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php [hunk 2]
```diff
@@ -61,8 +61,15 @@ public function addTransfer($paymentName, array $transferInformation)
             $transferInformation['creditorName']
         );
         $transfer->setBic($transferInformation['creditorBic']);
-        $transfer->setRemittanceInformation($transferInformation['remittanceInformation']);
-
+        
+        if(array_key_exists('creditorReferenceInformation', $transferInformation) &&
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: CGL (trying to follow PSR-2) space between if (

File: lib/Digitick/Sepa/DomBuilder/CustomerCreditTransferDomBuilder.php [hunk 3]
```diff
@@ -171,13 +171,24 @@ public function visitTransferInformation(TransferInformationInterface $transacti
         $creditorAccount->appendChild($id);
         $CdtTrfTxInf->appendChild($creditorAccount);
 
-        // remittance 2.98 2.99
-        if (strlen($transactionInformation->getRemittanceInformation()) > 0)
+
+        if($transactionInformation->getCreditorReferenceInformation())
```
Comment (monofone) [inline lib/Digitick/Sepa/DomBuilder/CustomerCreditTransferDomBuilder.php]: CGL (trying to follow PSR-2) space between if (

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php [hunk 4]
```diff
@@ -61,8 +61,15 @@ public function addTransfer($paymentName, array $transferInformation)
...
+            strlen($transferInformation['creditorReferenceInformation'])>5 ) {
+                
+            $transfer->setCreditorReferenceInformation($transferInformation['creditorReferenceInformation']); 
+        }else{
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: Is it really mutual exclusive that the remittance information can be set then there is no creditorReferenceInformation ?

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerDirectDebitFacade.php [hunk 5]
```diff
@@ -126,8 +126,8 @@ public function addTransfer($paymentName, array $transferInformation)
         if (isset($transferInformation['originalDebtorIban'])) {
             $transfer->setOriginalDebtorIban($transferInformation['originalDebtorIban']);
         }
-        if (isset($transferInformation['amendedDebtorAccount'])) {
-            $transfer->setAmendedDebtorAccount((bool) $transferInformation['amendedDebtorAccount']);
+        if (isset($transferInformation['amendedDebtorAgent'])) {
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerDirectDebitFacade.php]: This break BC is this useful ?

File: lib/Digitick/Sepa/TransferInformation/CustomerDirectDebitTransferInformation.php [hunk 6]
```diff
@@ -44,7 +44,7 @@ class CustomerDirectDebitTransferInformation extends BaseTransferInformation
     /**
      * @var bool
      */
-    protected $amendedDebtorAccount = false;
+    protected $amendedDebtorAgent = false;
```
Comment (monofone) [inline lib/Digitick/Sepa/TransferInformation/CustomerDirectDebitTransferInformation.php]: As stated above this breaks BC I follow you in that it makes sense but do you really want to introduce a BC break for this change ? I recommend to create a separate PR for this and we can make a newer version which will break BC and clean up some other stuff together with this

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php (hunk 4 above)
Comment (BorislavSabev) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: @monofone  they are not mutually-exclusive. In fact the CreditorReferenceInformation sits under the RemittanceInformation element!

The official docs are in [this MDR zipfile](https://www.iso20022.org/sites/default/files/documents/general/ISO20022MDR_PaymentsInitiation_2016_2017.zip). Inside this archive the document _ISO20022_MDRPart2_PaymentsInitiation_2016_2017_v1.pdf_ describes the structure of the whole **pain.001.001.vv** message.

The numbering of the elements in this same document further shows this. For a  reference to the info that shows this:
2.4.2.18        CreditTransferTransactionInformation (<CdtTrfTxInf>) [1..*] -> page: 31-32 
2.4.2.18.25     CreditorReferenceInformation (<CdtrRefInf>) [0..1] -> page: 66
2.4.2.18.25.2.3 RemittanceInformation (<RmtInf>) [0..1] -> page: 47

File: lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php (hunk 1 above)
Comment (SevenbitsIt) [inline lib/Digitick/Sepa/TransferFile/Facade/CustomerCreditFacade.php]: this ist only to make sure there is a number. usualy it is up to 25 digits long

**Comment by monofone** [comment]
Are you interested in this PR anymore I would be glad if you go to fix the issues ...

**Pull request closed without merging**